### ISSUE TITLE
Error when importing ts files from other ts files without extension

### ISSUE THREAD

**Issue by ocsi0520** [BjornTheProgrammer/bun-plugin-html#24]
Version: "bun-plugin-html": "^2.2.2"

Steps to reproduce:
1. `mkdir bun-plugin-html-hard-try`
2. `cd bun-plugin-html-hard-try`
3. `bun init -y`
4. `bun i -D bun-plugin-html`
5. create _src_ folder, and move index.ts into it (an _index.ts_ is already generated with a simple `console.log` statement)
6. create an _index.html_ under _src_
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script src="./index.ts" type="module"></script>
</body>
</html>
```
7. create _moduleB.ts_ and _moduleA.ts_ files in src folder:
```typescript
// src/moduleB.ts
export const getARandomNumber = (): number => Math.floor(Math.random() * 1000);
```

```typescript
// src/moduleA.ts
import { getARandomNumber } from './moduleB';

export type Result = {
  fails: number;
  successNumber: number;
}

export const runStrangeLogic = (): Result => {
  let fails = -1;
  let randomNum: number = NaN;
  while (randomNum < 700) {
    randomNum = getARandomNumber();
    fails++;
  }

  return {
    fails,
    successNumber: randomNum
  };
}
```

8. alter _index.ts_ file:
```typescript
// index.ts
import { runStrangeLogic } from './moduleA';

console.log(runStrangeLogic());
```
9. Create a _build.ts_ file in root folder
```typescript
// build.ts
import html from 'bun-plugin-html';

await Bun.build({
  entrypoints: ['src/index.html'],
  outdir: './dist',
  plugins: [ html() ],
});
```

10. modify _package.json_, by adding following code:
```json
"scripts": {
  "build": "bun build.ts"
}
```

11. `bun run build` and surprise it **does not** work
> [ error: File not found "<...>/src/moduleA"  at <...>/src/moduleA ]

12.   alter _build.ts_ file, so that it references _index.ts_ instead of _index.html_:
```typescript
// build.ts
import html from 'bun-plugin-html';

await Bun.build({
  entrypoints: ['src/index.ts'],
  outdir: './dist',
  plugins: [ html() ],
});
```
13. **surprise it works**

**Comment by BjornTheProgrammer** [comment]
Thank you so much for the report! I'll try to look into this problem later this week, since I have finals. I assume it has something to do with the custom resolver we use because of this issue in bun [here](https://github.com/oven-sh/bun/issues/15033). I've submitted a [pull request](https://github.com/oven-sh/bun/pull/15167) to fix that issue, but it still might be a while before it gets implemented.

I suspect that if you change [this](https://github.com/BjornTheProgrammer/bun-plugin-html/blob/main/src/index.ts#L353) to not need to be a directory, then this would work as expected though.

**Comment by BjornTheProgrammer** [comment]
@mindon I appreciate all of the work you have done on `bun-plugin-html`. If you are interested, I would be delighted to make you a maintainer.

**Comment by BjornTheProgrammer** [comment]
Should be fixed in `v2.2.3`
